src/library_pthread.js

```diff
@@ -1197,6 +1197,25 @@ var LibraryPThread = {
       worker.postMessage({'cmd' : 'processThreadQueue'});
     }
     return 1;
+  },
+
+  _emscripten_notify_proxying_queue: function(targetThreadId, currThreadId, mainThreadId, queue) {
```

Comment:
Can we locate this to a .js file of its own, e.g. `library_proxy_queue.js`?

Reply:
That sounds nice. I'll look into it.

Reply:
It turned out to be pretty simple to move this into its own JS file, but given that the implementation uses internal details of the pthread library (e.g. `PThreads.pthreads[targetThreadId]` and `pthread.worker`, and given that the existing `_emscripten_notify_thread_queue` will be removed in the next PR, I'm not sure there's much benefit to this after all.

Reply:
I think its small enough and closely coupled enough to belong here... and as you say we hope it will replace `scripten_notify_thread_queue` which is already here.